model/link_metadata.go

```diff
@@ -171,9 +171,9 @@ func (o *LinkMetadata) DeserializeDataToConcreteType() error {
 
 // FloorToNearestHour takes a timestamp (in milliseconds) and returns it rounded to the previous hour in UTC.
 func FloorToNearestHour(ms int64) int64 {
-	t := time.Unix(0, ms*int64(1000*1000))
+	t := time.Unix(0, ms*int64(1000*1000)).UTC()
 
-	return time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), 0, 0, 0, t.Location()).UnixNano() / int64(time.Millisecond)
+	return time.Date(t.Year(), t.Month(), t.Day(), t.Hour(), 0, 0, 0, time.UTC).UnixNano() / int64(time.Millisecond)
```

Comment:
This change is not strictly needed but I think it doesn't hurt to be extra explicit.